api/src/wfl/module/covid.clj

```diff
@@ -133,30 +203,15 @@
     (jdbc/execute! tx (concat [set-details] (map first src-exc-snk) [items]))
     items))
 
-(defn ^:private add-workload-record
-  "Use `tx` to create a workload `record` for `request` and return the id of the
-   new workload."
-  [tx request]
-  (letfn [(combine-labels [labels]
-            (->> (mapv request [:pipeline :project])
-                 (map str ["pipeline:" "project:"])
-                 (concat labels)
-                 set
-                 sort
-                 vec))]
-    (-> (update request :labels combine-labels)
-        (select-keys [:creator :watchers :labels :project])
-        (merge (select-keys (wfl/get-the-version) [:commit :version]))
-        (assoc :executor "" :output "" :release "" :wdl "" :uuid (UUID/randomUUID))
-        (->> (jdbc/insert! tx :workload) first :id))))
-
-;; TODO: validate the request before creating the workload
 (defn create-covid-workload [tx request]
   (let [set-pipeline "UPDATE workload
                       SET pipeline = ?::pipeline
                       WHERE id = ?"
         id           (add-workload-record tx request)
```

Comment:
Note that assigning `id` to `(add-workload-record tx request)` begins workload creation (a hint is that we pass the transaction).

Per the deleted comment, you want to validate source, executor and sink requests before creating the workload. I'd advise doing so before the `(let… )`.

Reply:
Done